docs/guides/requests_storage.md

```diff
@@ -0,0 +1,137 @@
+
+// Enqueue few requests
+await queue.addRequest({ url: 'http://example.com/aaa' });
+await queue.addRequest({ url: 'http://example.com/bbb' });
+await queue.addRequest({ url: 'http://example.com/foo/bar' }, { forefront: true });
+
+// Get requests from queue
+const request1 = await queue.fetchNextRequest();
+const request2 = await queue.fetchNextRequest();
+const request3 = await queue.fetchNextRequest();
+
+// Mark a request as handled
+await queue.markRequestHandled(request1);
+
+// If processing fails then reclaim the request back to the queue, so that it's crawled again
+await queue.reclaimRequest(request2);
+```
+
+To see how to use the request queue with a crawler, see the [Puppeteer Crawler](/docs/examples/puppeteer-crawler) example.
+
+## Request list
+
+The request list is not a storage per se - it represents the list of URLs to crawl that is stored in a run memory (or optionally in default [Key-Value Store](../guides/results-storage#key-value-store) associated with the run, if specified). The list is used for the crawling of a large number of URLs, when you know all the URLs which should be visited by the crawler and no URLs would be added during the run. The URLs can be provided either in code or parsed from a text file hosted on the web.
+
+Request list is created exclusively for the actor run and only if its usage is explicitly specified in the code. Its usage is optional.
+
+In Apify SDK, the request list is represented by the [`RequestList`](/docs/api/request-list) class.
+
+The following code demonstrates basic operations of the request list:
+
+```javascript
+// Prepare the sources array with URLs to visit and open the request list.
+// List name is used to persist the sources and the list state in the key-value store
+const sources = [
+    { url: 'http://www.example.com/page-1' },
+    { url: 'http://www.example.com/page-2' },
+    { url: 'http://www.example.com/page-3' },
+];
+const requestList = await Apify.openRequestList('my-list', sources);
+
+// Get number of requests added to the request list
+const requestsNumber = requestList.length();
+
+// Get number of handled requests
+const handledRequestsNumber = requestList.handledCount();
+```
+
+To see how to use the request list with a crawler, see the [Puppeteer with proxy](/docs/examples/puppeteer-with-proxy) example.
+
+## Which one to choose?
+
+When using Request queue - you would normally have several start URLs (e.g. category pages on e-commerce website) and then recursively add more (e.g. individual item pages) programmatically to the queue, it supports dynamic adding and removing of requests. No more URLs can be added to Request list after its initialization as it is static, URLs could not be removed from the list either.
+
+On the other hand, the Request queue is not optimized for adding or removing numerous URLs in a batch. Request list however could contain even millions of URLs, and it would take significantly less time to add them to the list.
```

Comment:
This sounds like the queue can't support millions of requests. It can, it just takes time to add them.

Reply:
Updated to better reflect the difference and make it clear that the difference is time consumed to add requests